ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -77,16 +80,20 @@
   private final StoreKeyFactory factory;
   private final File bloomFile;
   private final StoreMetrics metrics;
-  // an approximation of the last modified time.
-  private final AtomicLong lastModifiedTimeSec;
   private final AtomicInteger numberOfItems;
 
+  // an approximation of the last modified time.
+  private AtomicLong lastModifiedTimeSec;
   private MappedByteBuffer mmap = null;
   private IFilter bloomFilter;
   private int keySize;
   private int valueSize;
-  private Offset prevSafeEndPoint = null;
+  private short version;
```

Comment:
incase of persisted file, we have to populate by reading it from file. Hence can't be made final